core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java

```diff
@@ -308,34 +309,69 @@ public static long diffDate(@SqlType("varchar(x)") Slice unit, @SqlType(Standard
         return getDateField(UTC_CHRONOLOGY, unit).getDifferenceAsLong(DAYS.toMillis(date2), DAYS.toMillis(date1));
     }
 
+    /**
+     * Data tuple to store mapping between date time unit and date fraction extraction logic.
+     * This record is intended to be used for a specific use case only.
+     *
+     * @param unit date time unit represented as ASCII bytes
+     * @param dateTimeFieldProvider callback to extract fraction of date
+     */
+    private record DateTimeFieldProvider(byte[] unit, Function<ISOChronology, DateTimeField> dateTimeFieldProvider)
+    {
+    }
+
+    private static final DateTimeFieldProvider[] DATE_FIELDS = new DateTimeFieldProvider[] {
```

Comment:
usually we keep static fields at the top of the class

Reply:
Fixed